File: back/src/main/java/com/dev/ServiceHelp/dto/TicketDTO.java
package com.dev.ServiceHelp.dto;

import com.dev.ServiceHelp.enums.StatusTicket;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TicketDTO {
    private Long id;
    private String subject;
    private String description;
    private Instant registrationDate;
    private Instant dueDate;
    private StatusTicket statusTicket;
    private Instant completionDate;
    private TypeRequestDTO typeRequest;
    private SLADTO sla;
    private SolvingAreaDTO solvingArea;
    private CategoryTicketDTO categoryTicket;
    private UserSimpleDTO requester;
    private UserSimpleDTO technician;
    private UserSimpleDTO resolver;
    private Set<TicketHistoryDTO> ticketHistories;
    private Set<AttachmentDTO> attachments;
}

File: App/supportService.v4/src/models/ticketDTO.ts
import { AttachmentsDTO } from "./AttachmentsDTO";
import { CategoryTicketDTO } from "./CategoryTicketDTO";
import { UserDTO } from "./RequesterDTO";
import { SLADTO } from "./slaDTO";
import { SolvingAreaDTO } from "./solvingAreaDTO";
import { TicketHistoriesDTO } from "./TicketHistoriesDTO";
import { TypeRequestDTO } from "./typeRequestDTO";

export type TicketDTO = {
  id: number;
  subject: string;
  description: string;
  registrationDate: string;
  dueDate: string;
  statusTicket: string;
  completionDate: string;
  parentTicketId: number;
  typeRequest: TypeRequestDTO;
  sla: SLADTO;
  solvingArea: SolvingAreaDTO;
  categoryTicket: CategoryTicketDTO;
  requester: UserDTO;
  technician: UserDTO;
  resolver: UserDTO;
  ticketHistories: TicketHistoriesDTO[];
  attachments: AttachmentsDTO[];
};

export type TicketSimpleDTO = Omit<
  TicketDTO,
  "ticketHistories" | "attachments"
>;
